bluemira/equilibria/opt_problems.py

```diff
@@ -815,3 +819,270 @@ def get_state_figure_of_merit(
+        z_c: float
+            Vertical coordinate of the breakdown point
+        """
+        pass
+
+    @abc.abstractproperty
+    def breakdown_radius(self) -> float:
+        """
+        The radius of the breakdown zone.
+        """
+        pass
+
+    @abc.abstractmethod
+    def calculate_zone_points(self, n_points: int) -> Tuple[np.ndarray]:
+        """
+        Calculate the discretised set of points representing the breakdown zone.
+        """
+        pass
+
+
+class CircularZoneStrategy(BreakdownZoneStrategy):
+    """
+    Circular breakdown zone strategy.
+    """
+
+    def calculate_zone_points(self, n_points: int) -> Tuple[np.ndarray]:
+        """
+        Calculate the discretised set of points representing the breakdown zone.
+        """
+        x_c, z_c = self.breakdown_point
+        r_c = self.breakdown_radius
+        theta = np.linspace(0, 2 * np.pi, n_points - 1, endpoint=False)
+        x = x_c + r_c * np.cos(theta)
+        z = z_c + r_c * np.sin(theta)
+        x = np.append(x, x_c)
+        z = np.append(z, z_c)
+        return x, z
+
+
+class InboardBreakdownZoneStrategy(CircularZoneStrategy):
+    """
+    Inboard breakdown zone strategy.
+    """
+
+    @property
+    def breakdown_point(self) -> Tuple[float]:
+        r_c = self.breakdown_radius
+        x_c = self.R_0 - self.R_0 / self.A - self.tk_sol + r_c
+        z_c = 0.0
+        return x_c, z_c
+
+    @property
+    def breakdown_radius(self) -> float:
+        return 0.5 * self.R_0 / self.A
+
+
+class OutboardBreakdownZoneStrategy(CircularZoneStrategy):
+    """
+    Outboard breakdown zone strategy.
+    """
+
+    @property
+    def breakdown_point(self) -> Tuple[float]:
+        r_c = self.breakdown_radius
+        x_c = self.R_0 + self.R_0 / self.A + self.tk_sol - r_c
+        z_c = 0.0
+        return x_c, z_c
+
+    @property
+    def breakdown_radius(self) -> float:
+        return 0.7 * self.R_0 / self.A
+
+
+class InputBreakdownZoneStrategy(CircularZoneStrategy):
+    """
+    User input breakdown zone strategy.
+    """
+
+    def __call__(self, *args, **kwargs):
+        return self
+
+    def __init__(self, x_c, z_c, r_c):
+        self.x_c = x_c
+        self.z_c = z_c
+        self.r_c = r_c
+
+    @property
+    def breakdown_point(self) -> Tuple[float]:
+        return self.x_c, self.z_c
+
+    @property
+    def breakdown_radius(self) -> float:
+        return self.r_c
+
+
+class PremagnetisationCOP(CoilsetOptimisationProblem):
+    """
+    Coilset optimisation problem for the premagnetisation / breakdown phase.
+    """
+
+    def __init__(
+        self,
+        coilset: CoilSet,
+        breakdown_strategy: BreakdownZoneStrategy,
+        B_stray_max,
+        B_stray_con_tol,
+        n_B_stray_points,
+        optimiser: Optimiser = None,
+        max_currents=None,
+        constraints: List[OptimisationConstraint] = None,
+    ):
+        self.scale = 1e6  # current_scale
```

Comment:
This seems not hugely robust, say if we no longer use MA for whatever reason this will break, can this be at least configurable. The opposite also should be defined here instead of 1e-6 in the optimise method

Reply:
possible robust alternative: 

```
def __init__(..., scale="MA"):
    self.scale = raw_uc(1, scale, "A")
    self.inv_scale = raw_uc(1, "A", scale)

Reply:
More on this in #1027 I reckon. Trying to get rid of scale as much as possible so that it is set in only one place.